File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution { 
    public long solution(long n) {
        
        //double doubleN = (double)n;
        double rootN = Math.sqrt(n);
        
        //System.out.println(doubleN);
        System.out.println(rootN);
        long longN = (long)rootN;
        if(longN == rootN) return (longN + 1) * (longN + 1);
        else return -1;
        
        //boolean isSame = 
        
//         long answer = 0;
//         long count = 1;
//         while(n >= count){
//             if(n / count == count){
//                 answer = (count+1) * (count+1);
//                 return answer;
//             }
//             count++;
//         }
        
//         return -1;
        //return -1;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.js
function solution(n) {
    
    const input = String(n ** (1/2));
     
    if(input.includes('.')){
        return -1;
    }
    
    let answer = 0n;
    answer = (BigInt(input)+1n) ** 2n;     
        
    return answer;
}